### PR TITLE
[fix] Rearrange overlap depending in annotations count

### PR DESCRIPTION
**Pull request by KonstantinKorotaev**

### CODE REVIEW

File: label_studio/projects/models.py [hunk 1]
```diff
@@ -916,3 +924,8 @@ def remove_created_annotations_and_labels(self, annotations):
         self.created_annotations = created_annotations
         self.created_labels = labels
         self.save()
+
+
+Q_finished_annotations = Q(annotations__was_cancelled=False) & \
```
Comment (triklozoid) [inline label_studio/projects/models.py:929]: We already have `Q_finished_annotations` in `tasks.models`. I think we can add `Q(annotations__ground_truth=False)` to it, fix the only place where it is used, and import it there without defining a new one.

**Review by triklozoid — CHANGES_REQUESTED** [review]

File: label_studio/projects/models.py (hunk 1 above)
Comment (KonstantinKorotaev) [reply label_studio/projects/models.py]: Fixed by using Q from tasks.models

**Review by triklozoid — DISMISSED** [review]

File: label_studio/projects/models.py [hunk 2]
```diff
@@ -353,25 +353,33 @@ def update_tasks_states(
...
+        if left_must_tasks > 0:
+            tasks_with_max_annotations.update(overlap=max_annotations)
+            tasks_with_min_annotations = tasks_with_min_annotations.annotate(anno=Count('annotations')).order_by(
+                '-anno')
+            objs = []
+            for item in tasks_with_min_annotations[:left_must_tasks]:
+                item.overlap = max_annotations
+                objs.append(item)
+            Task.objects.bulk_update(objs, ['overlap'], batch_size=1000)
+            objs = []
```
Comment (KonstantinKorotaev) [inline label_studio/projects/models.py:375]: Change to one 1 bulk update

File: label_studio/projects/models.py [hunk 3]
```diff
@@ -353,25 +353,33 @@ def update_tasks_states(
...
-        elif tasks_with_overlap_count < new_tasks_with_overlap_count:
-            increase_by = new_tasks_with_overlap_count - tasks_with_overlap_count
-            tasks_without_overlap = self.tasks.filter(overlap=1)
-            increase_tasks = sample_query(tasks_without_overlap, increase_by)
-            increase_tasks.update(overlap=self.maximum_annotations)
-            tasks_with_overlap.update(overlap=self.maximum_annotations)
+        """
+        Rearrange overlap depending on annotations count in tasks
+        """
+        all_project_tasks = Task.objects.filter(project=self)
```
Comment (KonstantinKorotaev) [inline label_studio/projects/models.py:359]: Add comments

File: label_studio/projects/models.py [hunk 4]
```diff
@@ -353,25 +353,33 @@ def update_tasks_states(
...
+                item.overlap = max_annotations
+                objs.append(item)
+            Task.objects.bulk_update(objs, ['overlap'], batch_size=1000)
+            objs = []
+            for item in tasks_with_min_annotations[left_must_tasks:]:
+                item.overlap = 1
+                objs.append(item)
+            Task.objects.bulk_update(objs, ['overlap'], batch_size=1000)
+        else:
+            tasks_with_max_annotations.update(overlap=max_annotations)
```
Comment (KonstantinKorotaev) [inline label_studio/projects/models.py:381]: Add bulk_update_stats_project_tasks after tasks are updated

File: label_studio/projects/models.py (hunk 3 above)
Comment (KonstantinKorotaev) [reply label_studio/projects/models.py:359]: Fixed

File: label_studio/projects/models.py (hunk 4 above)
Comment (KonstantinKorotaev) [reply label_studio/projects/models.py:383]: Fixed

**Review by triklozoid — APPROVED** [review]

**Pull request closed without merging**